### PR TITLE
Update release-notes with upgrading steps to 0.9.x

### LINKED ISSUE #3497: Upgrade path from v0.8.0 to v0.9.0 is broken
**What happened**:

Running `kubectl apply --server-side -f https://github.com/kubernetes-sigs/kueue/releases/download/v0.9.0/manifests.yaml` has two failures:

```
Error from server (Invalid): CustomResourceDefinition.apiextensions.k8s.io "multikueueclusters.kueue.x-k8s.io" is invalid: status.storedVersions[0]: Invalid value: "v1alpha1": must appear in spec.versions
Error from server (Invalid): CustomResourceDefinition.apiextensions.k8s.io "multikueueconfigs.kueue.x-k8s.io" is invalid: status.storedVersions[0]: Invalid value: "v1alpha1": must appear in spec.versions
```

This results in crash-looping kueue-controller-manager:

```
$ k get pods -n kueue-system
NAME                                        READY   STATUS             RESTARTS     AGE
kueue-controller-manager-6f8ccbd7b8-ld6qp   2/2     Running            0            53s
kueue-controller-manager-74c44b9d78-mkdfl   1/2     CrashLoopBackOff   3 (6s ago)   48s
```

**What you expected to happen**:

v0.9.0 successfully installs into a cluster with v0.8.0 installed

**How to reproduce it (as minimally and precisely as possible)**:

1. Create kind cluster
2. `kubectl apply --server-side -f https://github.com/kubernetes-sigs/kueue/releases/download/v0.8.0/manifests.yaml`
3. `kubectl apply --server-side -f https://github.com/kubernetes-sigs/kueue/releases/download/v0.9.0/manifests.yaml`

**Anything else we need to know?**:

Uninstalling kueue then re-installing fresh mitigates the issue.  Is not an option for customers however because it would wipe their kueue installation

**Environment**:
- Kubernetes version (use `kubectl version`):

```
Client Version: v1.31.1
Kustomize Version: v5.4.2
Server Version: v1.31.1
```

- Kueue version (use `git describe --tags --dirty --always`): v0.9.0
- Cloud provider or hardware configuration: kind
- OS (e.g: `cat /etc/os-release`): 
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Update release notes.

#### Which issue(s) this PR fixes:

Fixes #3497

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: CHANGELOG/CHANGELOG-0.9.md [hunk 1]
```diff
@@ -26,6 +26,33 @@ Changes since `v0.8.0`:
 
   The QueueVisibility feature and its corresponding API was deprecated and will be removed in the v1beta2. Please use VisibilityOnDemand (https://kueue.sigs.k8s.io/docs/tasks/manage/monitor_pending_workloads/pending_workloads_on_demand/) instead. (#3110, @mbobrovskyi)
 
+## How to upgrade Kueue to 0.9.x
```
Comment (mimowo) [inline CHANGELOG/CHANGELOG-0.9.md:29]:
```suggestion
## Upgrading steps
```
nit, it feels a bit weird to start with "How" and not end with "?". Also Kueue is redundant here.

Comment (mbobrovskyi) [reply CHANGELOG/CHANGELOG-0.9.md]: Done

**Comment by mimowo** [comment]
/lgtm
/approve
/cherry-pick release-0.9
So that we can update the release notes for the release.

**Comment by mimowo** [comment]
/retitle Update release-notes with upgrading steps to 0.9.x

**Pull request merged**